### PR TITLE
feat: regression tests for optional call

### PR DESCRIPTION
**Pull request by bcoe**
Adds regression tests for #2414, CC: @jridgewell.

fixes: #2414

### CODE REVIEW

File: test/language/expressions/optional-chaining/new-target-optional-call.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+let called = false;
+function Base() {
+  called = true;
+}
+function Foo(blerg) {
+  new.target?.();
+}
+Reflect.construct(Foo, [], Base);
+assert.sameValue(called, true);
```
Comment (jridgewell) [inline test/language/expressions/optional-chaining/new-target-optional-call.js]: We should also check that the `this` context is correct?

Comment (bcoe) [reply test/language/expressions/optional-chaining/new-target-optional-call.js]: @jridgewell which `this` context:

```js
function Base() {
  return this;
}
function Foo(blerg) {
  return new.target();
}
const result = Reflect.construct(Foo, [], Base);
```

☝️ the `this` context of `Base` in `v8` appears to be the `global` object in this example, i.e.,

```
result istanceof Base // is false.
```

```js
function Foo(blerg) {
  return new.target?.();
}
```

behaves similarly.

Comment (jridgewell) [reply test/language/expressions/optional-chaining/new-target-optional-call.js]: It should be the global in sloppy mode and `undefined` in strict.

Comment (bcoe) [reply test/language/expressions/optional-chaining/new-target-optional-call.js]:
```js
let called = false;
function Base() {
  called = true;
  return this;
}
function Foo(blerg) {
  return new.target?.();
}
assert(Reflect.construct(Foo, [], Base) === this);
assert.sameValue(called, true);
```

Comment (jridgewell) [reply test/language/expressions/optional-chaining/new-target-optional-call.js]: Yup. Then for `super.method?.()`, too (which has a much easier `this` context to understand).

Comment (leobalter) [reply test/language/expressions/optional-chaining/new-target-optional-call.js]: just adding a quick tip: `var t = (function() { return this; })();` so you reuse the default this value for sloppy and strict mode.

Comment (bcoe) [reply test/language/expressions/optional-chaining/new-target-optional-call.js]: @leobalter oh, If I understand, we already run tests both in both `sloppy` and `strict` mode, so I would be able to simply have one test:

```js
const t = (function() { return this; })();
assert(Reflect.construct(Foo, [], Base) === t);
```

Comment (leobalter) [reply test/language/expressions/optional-chaining/new-target-optional-call.js]: assuming you want the global object, yes.

For consistency, you can also load a harness file, adding to the yaml meta:

```yaml
includes: [fnGlobalObject.js]
---*/
```

so in your js you can use: `const global = fnGlobalObject();`

**Comment by bcoe** [comment]
@jridgewell @leobalter take a look, sorry it took me a few days to get back to this.

**Pull request merged**